### ISSUE TITLE
Not always receiving message notifications when phone is idle - perhaps Doze related?

### ISSUE THREAD

**Issue by DivineOmega** [wireapp/wire-android#254]
Unfortunately, I'm experiencing an issue with Wire on Android 7 (Nexus 6P) in which I do not always receive message notifications. However, the messages are present and marked as unread when I open the Wire app.

I have noticed this usually occur when I have not used my phone in a short while. I'm wondering if this has anything to do with the Doze feature present in newer versions of Android? In most cases, when an app has been 'dozed' I seem to get the notification after a delay or near immediately if I resume using my phone.

I certainly don't mind having delayed notifications, but not receiving the notifications at all when my phone has been idle affects usability quite severly.

I'm happy to provide any further information if it would be useful. Due to the requirement to leave the device idle, this is a pretty difficulty issue to reliable replicate in a typical environment.

**Comment by someoneEsle** [comment]
I don't have this problem on my Nexus 5X with Android 7

**Comment by charafau** [comment]
I also have problem with notifications. (Nexus 6 with Android 7.0)

**Comment by someoneEsle** [comment]
I think the developers may use a debug log captured right after opening the app -- if there are any delayed messages when you open it.
Next time you experience this issue when you open Wire, go to settings>advanced and capture a debug log. Hopefully the developers will find it useful :)

**Comment by levinus** [comment]
Actually I have these problems from time to time with mit Nexus 5X on Android 7 too.

**Comment by cwmke** [comment]
Maybe adding an option in settings to have a permanent notification in the tray like many other apps do would provide more reliable notifications.

I know that both Conversations as well as Riot provide this option.

**Comment by ColinLeverger** [comment]
I have the same issue here. (One Plus 3, Android 6.0.1)
I will put Wire app on Doze whitelist (so, this app won't be "Dozed" anyway) and see if it works better with that.

**Comment by someoneEsle** [comment]
I think the last update (which came out a couple of days ago) should have fixed this. Are you still experiencing it?

**Comment by ColinLeverger** [comment]
No, it is much better for a couple of days! I do not know if it is the update or the undoze of the app, though.

**Comment by someoneEsle** [comment]
I think that in your specific situation, undozing the phone was necessary. However, the latest update did fix bugs where you would not receive notifications, so I think this issue has been resolved. Let's hear what the owner of the issue has to say!

**Comment by savvakism** [comment]
This issue has not been resolved yet. I have sent a email with video example. This issue appears more than once in my android phone.

**Comment by deanrobertcook** [comment]
Hi guys, just to keep you all updated, we've been experiencing some issues with (very particular) users not receiving GCM notifications. We would try to detect these failures and fall back to a constant WebSocket connection. Problem is that the fallback mechanism had some issues and was a little bit aggressive in detecting if GCM had failed, so that affected many more people than necessary. The latest release tried to make this fallback a bit less aggressive, in the hope that would be enough.

There will be a second round of fixes coming very soon, and if the WebSocket fallback proves to be stable and doesn't drain the device's battery in our internal testing, then perhaps we'll use it permanently (those who don't have google play services on their devices, for example, have to rely on this connection anyway).

**Comment by peterpanondrugs** [comment]
After i opened WIRE to see if someone wrote me, i saw a message delivered 6hrs before the startup of WIRE. I had no notification on my screen.
I've set notifications in system (Android 6.0.1) for WIRE as prio mode. But still nothing.

[wire_debug_report.txt](https://github.com/wireapp/wire-android/files/652918/wire_debug_report.txt)

**Comment by mcg** [comment]
I, well actually my family of 3 Nexus 6P's on Android 7.1.1 and latest Wire, have unreliable notifications similar to what is described here. A persistent WebSocket connection worries me somewhat for battery drain and that Wire might end up with a huge background task running. Hopefully this can be fixed.

**Comment by gusstav** [comment]
Well, I have an Huawei Elite P8 with lollipop android, my problem is that I don´t receive sounds or any notification while Wire is in inactive mode. I wrote to Wire Support but they don´t fix the issue.

**Comment by someoneEsle** [comment]
@gusstav have you looked at #253 ?

**Comment by kozality** [comment]
So I'm going to give this a shot, and exempt Wire from Power Saving. I'll see if I notice any improvements over the next several days:
https://www.cnet.com/how-to/stop-specific-apps-from-using-doze-on-android-6-0-marshmallow/

**Comment by mhham** [comment]
Hi guys. I had the same issue with a Honor 5C Emui 4.1. Seems like exempting Wire from Power Saving does the trick.
Many of my relatives don't receive notifications on different devices though ..

**Comment by ConorIA** [comment]
I have turned off battery optimization for Wire on my device, and the messages usually get through. However, a few of my contacts will often tell me that they never saw my messages. In at least two cases, I have double-checked and their phones are using a version of Android that didn't include the "battery optimization" feature. At least one contact has suggested (God forbid) that we switch to using Whatsapp .. which seems to imply that she gets Whatsapp message notifications no problem (though I do not, probably because I allow Whatsapp to be "battery optimized").

**Comment by someoneEsle** [comment]
@ConorIA What phones are your friends on? Might be that they have another setting in the Network/data section that prevents wire from sending data while in background.

**Comment by ConorIA** [comment]
@someoneEsle, one is a Blu Vivo XL. The other is a Huawei "something or other". I think both are on Android 6.0 by now.

**Comment by mhham** [comment]
A small update, a lot of my friends (iPhone 5S, iPhone 6, Samsung G4, Nexus 5) still do not receive notifications for new messages and incoming audio/video calls.
Seems like the problem is still not solved ...

**Comment by vith** [comment]
wire-android is very often not clearing notifications on the phone for messages I already read on desktop until I unlock the phone. Same bug?

Android 7.1.1

**Comment by Simplenuity** [comment]
Also experiencing this problem. Most of the times I've Wire parallel running on Android device (only to actually hear the notification) and on Linux Desktop (via app) I then use for message exchange:
Currently: 
Android 5.1.1 with Wire 2.30.321
Telegram message notification 100% reliable, Wire maybe around 50-70%. I made sure to close the window of Wire Linux Desktop app or at least don't have it activated.

**Comment by Simplenuity** [comment]
Additional observation: At least sometimes the Android app only notifies of messages that arrived a while ago after I've sent back the first reply via the desktop app.
Tried to create a bug report on Android device, but got: "Unable to open log device 'main'"

**Comment by Simplenuity** [comment]
And another one: With the Wire Android app (device on WiFi) being the only app logged into the account (i.e. desktop app was shut down) the notifications (and messages, see below) didn't come through. Only when I opened Wire hours later (after I saw new messages on the restarted desktop app) the Wire Android app actually downloaded the messages. I could observe them showing up one by one.
On the Android device Wire has all the permissions including "Wake up", has been excluded from any power saving app, specifically Greenify. And, sometimes notifications/messages coming through without problem, even with the phone haven't been touched for a while. As mentioned before, on the same device Telegram runs without any issue. 
Not sure what you need from my end to troubleshoot this. Please let me know.

**Comment by axaxs** [comment]
I'll add a 'me too', across multiple devices.  Tested with Axon 7, Honor 8, and Le Pro 3.  This is easily the most frustrating bug as it really makes the app unreliable/unusable unless you just open the app every few minutes.  I do not believe it to be power setting related - every other messenger I've used notifies fine.

**Comment by Trevoke** [comment]
This has happened occasionally for me and a friend of mine as well.

**Comment by lingfish** [comment]
I too am getting this behaviour with friends and workmates. Seems to work great on my devices though.  Do the devs need anything from us with this issue?  As I concur with @axaxs -- easily the best messaging app, if it actually 100% worked.

**Comment by beserkhobo** [comment]
Any testing we can do to help the issue with notifications not being received on some Android devices?

**Comment by Larx** [comment]
I have the same situation:
Galaxy S7, LineageOS, MicroG - everything OK
Galaxy S4 mini, LineageOS, MicroG - many notifications are simply missed!

However, as in MicroG you can see when the last message was received, the S4mini definitely receives the messages. Only Wire does not notice it?!

**Comment by deValmont** [comment]
I join the list
No notifications at all
Huawei Mate 9 Pro

**Comment by AUrhino** [comment]
Still having issues with notifications not being shown.
I can see a notification count on the wire app but not shown on the main Andriod screen.
Would you like some logs or screenshots showing this?

**Comment by someoneEsle** [comment]
Hi everyone, please have a look at this: https://github.com/wireapp/wire-android/issues/1087 :) Stay tuned!

**Comment by tokariu** [comment]
I had the same problem as OP mentioned.

I'm currently on a LG G3 + Lineageos 14.1 + microG, where microG is disabled and Wire *has* to use websockets. I'm only using Wifi no mobile data.

The problem was again, that after about 1 min idleness the phone doesn't receive any messages and notifications. 
I realizied that the problem doesn't seem to be Wire related anymore, because when I touch the phone again I see that it instantly tries to re-establish the wifi connection.
Though, it should have never lost the wifi connection, because I set the android wifi settings to "keep wifi activated on standby - always" (or the like). 
I googled around and found issues with this Android Wifi setting. It does not work as expected for many users. They all have set it to "keep wifi activated on standby - always" but it's not active during standby. 
Some guy found a solution that worked for me. The error is obviously that android is not applying the wifi setting. So what you have to do is set the setting from "always" to "never" and apply it by leaving the menu. then get back and change the setting back to "always" and leave the menu again. that way android seems to finally apply the wifi setting and the phone keeps the wifi connection while standby is on.
Wire couldn't get messages because android didn't apply the settings correctly. no wifi in standby meant no messages for wire while idle.

so to sum it up again:
if you want a working Wire with websockets you have to do:

- change the wifi settings "keep wifi connection while standby"
- apply "never" leave the menues (important!)
- apply "always" leave the menu again.

it is also important to disable battery savings for the Wire app, otherwise the app gets in a sleep state shortly after standby and messages won't be received either. There is an option under Battery/Akku almost hidden, where you can set apps to be "not optimized" for battery savings - wire needs to be "not optimized" to work.

**Comment by hub-ch** [comment]
https://github.com/wireapp/wire-android/issues/777#issuecomment-466763316

**Comment by marcoconti83** [comment]
Hi all,
in case it was missed, there's a checklist to go through if you notifications are still acting up: https://support.wire.com/hc/en-us/articles/203624294-What-can-I-do-if-I-m-not-getting-any-notifications-on-Android-

**Comment by marcoconti83** [comment]
Hi all,
there are a lot of duplicated issues covering this same problem (or a very similar one), let's unify all conversations and keep it going here: #777 . Closing this one and continuing there.
